### ISSUE TITLE
Last digit missing in bbox coordinate of changeset JSON call

### ISSUE THREAD

**Issue by nrenner** [zerebubuth/openstreetmap-cgimap#363]
The `minlon` and `maxlon` properties in the JSON response of the metadata API call for [changeset 146899741](https://www.openstreetmap.org/changeset/146899741) ([node 11572546705 v1](https://www.openstreetmap.org/node/11572546705/history/1)) are missing the last digit, the XML response has it:

`Actual  `: `-117.067166 ` - https://www.openstreetmap.org/api/0.6/changeset/146899741.json
`Expected`: `-117.0671669` - https://www.openstreetmap.org/api/0.6/changeset/146899741

This was noticed because the adiff query with that bbox in OSMCha fails to load the node, as [reported](https://osmus.slack.com/archives/C04Q8R2G17E/p1706907571545059) on Slack.

**Comment by Woazboat** [comment]
Posting the full response data here for future reference:

```json
{"version":"0.6","generator":"CGImap 0.8.10 (2938081 spike-07.openstreetmap.org)","copyright":"OpenStreetMap and contributors","attribution":"http://www.openstreetmap.org/copyright","license":"http://opendatacommons.org/licenses/odbl/1-0/","elements":[{"type":"changeset","id":146899741,"created_at":"2024-01-31T11:17:34Z","closed_at":"2024-01-31T11:17:36Z","open":false,"user":"newportequipmentwa1","uid":20953528,"minlat":48.1730392,"minlon":-117.067166,"maxlat":48.1730392,"maxlon":-117.067166,"comments_count":0,"changes_count":1,"tags":{"changesets_count":"1","comment":"Excavating Contractor","created_by":"iD 2.27.3","host":"https://www.openstreetmap.org/edit","imagery_used":"Bing Maps Aerial","locale":"en-US"}}]}
```

```xml
<osm version="0.6" generator="CGImap 0.8.10 (2351905 spike-07.openstreetmap.org)" copyright="OpenStreetMap and contributors" attribution="http://www.openstreetmap.org/copyright" license="http://opendatacommons.org/licenses/odbl/1-0/">
<changeset id="146899741" created_at="2024-01-31T11:17:34Z" closed_at="2024-01-31T11:17:36Z" open="false" user="newportequipmentwa1" uid="20953528" min_lat="48.1730392" min_lon="-117.0671669" max_lat="48.1730392" max_lon="-117.0671669" comments_count="0" changes_count="1">
<tag k="changesets_count" v="1"/>
<tag k="comment" v="Excavating Contractor"/>
<tag k="created_by" v="iD 2.27.3"/>
<tag k="host" v="https://www.openstreetmap.org/edit"/>
<tag k="imagery_used" v="Bing Maps Aerial"/>
<tag k="locale" v="en-US"/>
</changeset>
</osm>
```

It seemingly looks like only negative lat/lon values are affected. `"minlat":48.1730392` has the full 7 digits while `"minlon":-117.067166` only has 6.

**Comment by Woazboat** [comment]
The json formatter uses a 12 byte buffer to convert the `double` value to string, which is not enough to hold the full string when it has a minus sign in front so the last digit gets cut off.
https://github.com/zerebubuth/openstreetmap-cgimap/blob/master/src/json_writer.cpp#L115-L116

**Comment by mmd-osm** [comment]
I released version 0.9.0 earlier today and triggered the Ubuntu package build on https://launchpad.net/~mmd-osm/+archive/ubuntu/cgimap-900-jammy-test2/+packages

Still pending are local tests that the resulting package is working ok, then we'll raise an issue for osm operations to deploy the new version.

**Comment by mmd-osm** [comment]
Fix is in production now.
